### DISCUSSION TITLE
[월드 1.1.5] 멀티플레이 회전 동기화 문제

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#598] [5. Q&A]
월드 1.1.5 입니다 

멀티 플레이개발시 회전 동기화가 잘 안되는거 같네요 
터치 패드사용해서 이동중에 타겟을 바라보게 만들었습니다 
이때 LateUpdate에서 타겟을 찾아 바라보게 만들었는데요 
아래 영상과 같이 자신의 화면에서는 잘 회전되는데 
상대방 화면에서는 회전이 안됩니다 
무엇이 문제 일까요?
그냥 제 로직에 문제일까요? 
만들면 만들수록 어렵네요... 후 

<왼쪽이 상대방                         >오른쪽이 내화면 

https://user-images.githubusercontent.com/50188795/162168760-5ec73c45-1345-4202-9857-8ba64d96834e.mp4

아래는 LateUpdate에서 타겟을 바라보게 만드는 부분 
![image](https://user-images.githubusercontent.com/50188795/162168914-8297f466-296e-44ab-96e6-4926705ed4e4.png)

SendTransform 부분 
![image](https://user-images.githubusercontent.com/50188795/162169067-b733abd2-2cf5-40ea-92f3-8f014a0210c7.png)

아래는 서버 
![image](https://user-images.githubusercontent.com/50188795/162169172-3cc4a6e2-fcaa-4261-bc9b-07040d46a3d8.png)

아래는 스키마 
![image](https://user-images.githubusercontent.com/50188795/162169273-bf1efaa1-bdca-4895-ba1e-f7aee8e5de57.png)

**Comment by Joyounwoo** [comment]
타 유저의 제페토 캐릭터는 MoveToPosition() 함수를 통해 이동하게 됩니다.
이 때, 캐릭터 이동 시 해당 좌표 방향으로 캐릭터 각도를 변경하게 됩니다.
여기서 문제가 발생합니다.
캐릭터의 회전 값을 전송하여도, 이내 제페토 캐릭터에서 이동 방향으로 회전하기에 캐릭터 회전 값을 전송하여도 잠깐잠깐 회전이 바뀌고, 다시 이동하는 방향으로 회전하게 됩니다.
이 로직을 수정하셔야 의도하신 대로 동작할 것으로 보입니다.

**Reply by smilejsu82** [reply]
감사합니다 덕분에 영감을 얻어 해결했습니다 

https://user-images.githubusercontent.com/50188795/163105423-9ef71d96-dabf-4f4b-bbe9-af755f553abc.mp4
